File: csharp/FantasyBattle/PlayerTest.cs
using System.Collections.Generic;
using System.Text;
using Moq;
using Xunit;

namespace FantasyBattle
{
    public class PlayerTest
    {

        // choose this one if you are familiar with mocks
        [Fact(Skip = "Test is not finished yet")]
        public void DamageCalculationsWithMocks()
        {
            var inventory = new Mock<Inventory>();
            var stats = new Mock<Stats>();
            var target = new Mock<SimpleEnemy>();

            var damage = new Player(inventory.Object, stats.Object).CalculateDamage(target.Object);
            Assert.Equal(10, damage.Amount);
        }

        // choose this one if you don't like mocks  :o)
        [Fact]
        public void DamageCalculations()
        {
            var excalibur = new BasicItem("", 10, 1.0f);
            var shield = new BasicItem("", 0, 1.4f);
            var boots = new BasicItem("", 0, 0.1f);
            var helmet = new BasicItem("", 0, 1.2f);
            var plate = new BasicItem("", 0, 1.4f);
            var equipment = new Equipment(excalibur, shield, boots, helmet, plate);
            var inventory = new Inventory(equipment);
            var stats = new Stats(0);
            var armor1 = new SimpleArmor(5);
            var buffs = new List<Buff>() { new BasicBuff(1f, 1f) };
            var target = new SimpleEnemy(armor1, buffs);

            Damage damage = new Player(inventory, stats).CalculateDamage(target);
            Assert.Equal(41, damage.Amount);
        }
    }
}

File: kotlin/src/test/kotlin/codingdojo/PlayerTest.kt
package codingdojo

import com.nhaarman.mockitokotlin2.mock
import org.junit.Ignore
import org.junit.Test
import kotlin.test.assertEquals

class PlayerTest {

    // choose this one if you are familiar with mocks
    @Ignore("Test is not finished yet")
    @Test
    fun damageCalculationsWithMocks() {
        val inventory: Inventory = mock()
        val stats: Stats = mock()
        val target: SimpleEnemy = mock()

        val damage = Player(inventory, stats).calculateDamage(target)

        assertEquals(10, damage.amount)
    }

    // choose this one if you are not familiar with mocks
    // @Ignore("Test is not finished yet")
    @Test
    fun damageCalculations() {
        val excalibur = BasicItem("", 10, 1.0f)
        val shield = BasicItem("", 0, 1.4f)
        val boots = BasicItem("", 0, 0.1f)
        val helmet = BasicItem("", 0, 1.2f)
        val plate = BasicItem("", 0, 1.4f)
        val equipment = Equipment(leftHand = excalibur, rightHand = shield, head = boots, feet = helmet, chest = plate)
        val inventory = Inventory(equipment)
        val stats = Stats(0)
        val armor1 = SimpleArmor(5)
        val buffs = listOf<Buff>(BasicBuff(1f, 1f))
        val target = SimpleEnemy(armor = armor1, buffs = buffs)

        val damage = Player(inventory, stats).calculateDamage(target)

        assertEquals(45, damage.amount)
    }

    @Test
    fun damageCalculations2() {
        val excalibur = BasicItem("", 10, 1.0f)
        val shield = BasicItem("", 0, 1.4f)
        val boots = BasicItem("", 0, 0.1f)
        val helmet = BasicItem("", 0, 1.2f)
        val plate = BasicItem("", 0, 1.4f)
        val equipment = Equipment(leftHand = excalibur, rightHand = shield, head = boots, feet = helmet, chest = plate)
        val inventory = Inventory(equipment)
        val stats = Stats(0)

        val target = Player(inventory, stats)

        val damage = target.calculateDamage(target)

        assertEquals(0, damage.amount)
    }
}
